app/src/main/java/org/kinecosystem/kinit/view/customView/QuizAnswerView.kt

```diff
@@ -0,0 +1,150 @@
+import java.util.*
+import javax.inject.Inject
+
+
+class QuizAnswerView : ConstraintLayout {
+
+    interface OnSelectionListener {
+        fun onAnimComplete()
+        fun onAnswered(answer: Answer?)
+    }
+
+    @Inject
+    lateinit var scheduler: Scheduler
+
+    private val DELAY_AFTER_ANIM = 1500L
+    private val MAX_PARTICLES = 40
+    private val MIN_ANIM_PARTICLE_DURATION = 3000L
+    private val frameLayoutParam: FrameLayout.LayoutParams by lazy {
+        val param = FrameLayout.LayoutParams(LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+        param.gravity = Gravity.CENTER
+        param
+    }
+    private val particlesRes by lazy {
+        listOf(R.drawable.big_purple_circle, R.drawable.big_rectangle_turkiz, R.drawable.big_yellow_triangle, R.drawable.small_purple_circle, R.drawable.small_rectangle_turkiz, R.drawable.small_yellow_triangle)
+    }
+    private val rand by lazy { Random() }
+
+
+    lateinit var listener: OnSelectionListener
+    var isCorrect: Boolean = false
+    var reward: Int = 0
+        set(value) {
+            reward_text.text = "+$value KIN"
+        }
+
+    var answer: Answer? = null
+        set(value) {
+            answer_text.text = value?.text
+            answer_feedback.text = value?.text
+            field = value
+        }
+
+    constructor(context: Context) : super(context) {
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        init(context)
+    }
+
+    private fun init(context: Context) {
+        KinitApplication.coreComponent.inject(this)
+        val inflater = context
+                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        val view = inflater.inflate(R.layout.quiz_answer_layout, this, true)
+        setOnClickListener {
+            listener.onAnswered(answer)
+            if (isCorrect) {
+                showCorrect(true)
+            } else {
+                showWrong()
+            }
+        }
+    }
+
+    fun showCorrect(userClicked: Boolean = false) {
+        if (isCorrect) {
+            answer_feedback.background = resources.getDrawable(R.drawable.answer_bg_correct)
+            if (userClicked) {
+                //make the view last in parent
+                val container: ViewGroup = parent as ViewGroup
+                container.removeView(this)
+                container.addView(this)
+                animateBackground()
+                animateParticles()
+                animateReward()
+            } else {
+                animateBackground(true)
+            }
+        }
+    }
+
+    private fun animateReward() {
+        reward_text.y = reward_text.height.toFloat()
+        reward_text.alpha = 1f
+        reward_text.animate().setStartDelay(1100).translationY(0f).setInterpolator(OvershootInterpolator(3f)).setDuration(600).withEndAction {
+            answer_text.postDelayed({ listener.onAnimComplete() }, 2000)
```

Comment:
Might cause problems if the user navigates away before the time is up (2 seconds).
I would check if the listener exists before invoking `onAnumComplete()` and add option to remove the listener according to the appropriate lifecycle.

Reply:
i tried to test it make the animation long and navigate backed or keep in background it worked as expected